receiver/sqlserverreceiver/scraper.go

```diff
@@ -299,3 +327,272 @@ func (s *sqlServerScraperHelper) recordDatabaseStatusMetrics(ctx context.Context
 
 	return errors.Join(errs...)
 }
+
+func (s *sqlServerScraperHelper) recordDatabaseQueryTextAndPlan(ctx context.Context, topQueryCount uint) (plog.Logs, error) {
+	// Constants are the column names of the database status
+	const dbPrefix = "db."
+	const totalElapsedTime = "total_elapsed_time"
+	const rowsReturned = "total_rows"
+	const totalWorkerTime = "total_worker_time"
+	const queryHash = "query_hash"
+	const queryPlanHash = "query_plan_hash"
+	const logicalReads = "total_logical_reads"
+	const logicalWrites = "total_logical_writes"
+	const physicalReads = "total_physical_reads"
+	const executionCount = "execution_count"
+	const totalGrant = "total_grant_kb"
+	const queryText = "query_text"
+	const queryPlan = "query_plan"
```

Comment:
```suggestion
	const totalElapsedTime = "sqlserver.total_elapsed_time"
	const rowsReturned = "db.response.returned_rows"
	const totalWorkerTime = "sqlserver.total_worker_time"
	const queryHash = "sqlserver.query_hash"
	const queryPlanHash = "sqlserver.query_plan_hash"
	const logicalReads = "sqlserver.total_logical_reads"
	const logicalWrites = "sqlserver.total_logical_writes"
	const physicalReads = "sqlserver.total_physical_reads"
	const executionCount = "sqlserver.execution_count"
	const totalGrant = "sqlserver.total_grant_kb"
	const queryText = "db.query.text"
	const queryPlan = "sqlserver.query_plan"
```

Could we follow the common attributes from semantic convention https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/database-spans.md#common-attributes?

For SQL server specific attributes, we can add `sqlserver` prefix to it to prevent attribute conflict in the future.

Reply:
These are simply column names in the returned MSSQL rows. The actual attributes will include the `db.` prefix.

Regarding the semantic convention, I noticed that it primarily applies to spans and may not be directly relevant here. While we can certainly follow the convention, I believe it could create a cognitive burden for users when reading these attributes.

For example, if a user encounters an attribute like `db.response.returned_rows`, they might need to determine its meaning by searching in one of three places:

- The OpenTelemetry documentation
- The MSSQL documentation
- Both

With a name like `db.response.returned_rows`, they may only fully understand its meaning after searching both sources. However, by preserving the original column names from the existing MSSQL tables, users can easily look up their meanings directly in the MSSQL documentation.

Reply:
I think we have two issues here,

First is the possibility of **attribute type conflict**. If we use specific database values with a common prefix, say `db.`, this may lead to this problem sooner or later. For instance, a `query_hash` type in SQL server is a string, but it may be an integer in other databases. By using `db.query_hash`, we might end up having both string and integer types in the log/event store. Usually, the log/event store will expect data to have stable types. When data type conflict happens, depending on the implementation details, data missing or data corruption might happen. This is something we should avoid.

---

Second is the exact meaning of the attribute name.

For every attribute that is listed in the semantic convention, I think they have a concrete meaning of what it is and what it means. I don't think use `db.query.text` would bring confusion to users, as it is used widely in metrics and logs. And, its name does not include `response`, which seems designed for response type data. 

For `db.response.returned_rows`, I am ok to use `sqlserver.total_rows`, as the stability level of `db.response.returned_rows` is `Development`.

As for the `sqlserver` prefix, it is a very clear sign to users that this is a specific SQL server value. For users, it is better to check the SQL server documentation to understand the meaning.

If we use just the `db` prefix, this means it is a common value, not something special to a certain database system. If the meaning of the value is not clear, the OTel community is responsible for defining it.

---

About your concern,


> I noticed that it primarily applies to spans and may not be directly relevant here

I should point to the DB attribute registry page, https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/db.md, which does not bond to a certain signal.

> With a name like db.response.returned_rows, they may only fully understand its meaning after searching both sources.

If we use the `db` prefix, the semantic convention should already have the concrete meaning for the value; if not, then we should not use the `db` prefix.

Reply:
updated to use `sqlserver.` prefix and  `db.query.text`